### PR TITLE
remove uid when href exists

### PR DESCRIPTION
**Pull request by AngryBerryMS**
1. Remove uid in reference-toc (top level toc) when href is specified.
Although these uids don't affect the final pages, it will cause some xref-not-found warnings.
2. When the href of a service-page item in TopLevelTOC specifies an existing file (eg. .md file), should not generated itemNmae.yml file again.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6641)

### CODE REVIEW

**Comment by live1206** [comment]
some test to cover the `xref-not-found` warnings not thrown?

**Comment by AngryBerryMS** [comment]
> some test to cover the `xref-not-found` warnings not thrown?

No "xref-not-found" warning should be thrown for no invalid uid now.

File: src/docfx/build/toc/ServicePageGenerator.cs [hunk 1]
```diff
@@ -34,12 +34,55 @@ public void GenerateServicePageFromTopLevelTOC(TableOfContentsNode node, List<Fi
...
+
+            bool shouldGeneratedServicePage;
+            if (string.IsNullOrEmpty(node.Href.Value))
+            {
+                shouldGeneratedServicePage = true;
+            }
+            else
+            {
+                shouldGeneratedServicePage = !File.Exists(nodeHrefFullPath);
+            }
```
Comment (live1206) [inline src/docfx/build/toc/ServicePageGenerator.cs:50]:
```suggestion
            var shouldGeneratedServicePage = string.IsNullOrEmpty(node.Href.Value) || !File.Exists(nodeHrefFullPath);
```

File: src/docfx/build/toc/ServicePageGenerator.cs [hunk 2]
```diff
@@ -34,12 +34,55 @@ public void GenerateServicePageFromTopLevelTOC(TableOfContentsNode node, List<Fi
...
+                    GenerateServicePageFromTopLevelTOC(item, results, $"{directoryName}");
+                }
+                else
+                {
+                    GenerateServicePageFromTopLevelTOC(item, results, $"{directoryName}/{filename}");
+                }
+            }
+        }
+
+        private void GeneratedServicePageForItem(
```
Comment (live1206) [inline src/docfx/build/toc/ServicePageGenerator.cs:75]:
```suggestion
        private void GenerateServicePageForItem(
```

File: src/docfx/build/toc/ServicePageGenerator.cs [hunk 3]
```diff
@@ -34,12 +34,55 @@ public void GenerateServicePageFromTopLevelTOC(TableOfContentsNode node, List<Fi
...
+            TableOfContentsNode node,
+            string directoryName,
+            string filename,
+            string referenceTOCFullPath,
```
Comment (live1206) [inline src/docfx/build/toc/ServicePageGenerator.cs:79]: 'referenceTOCFullPath' not used in the method?

File: src/docfx/build/toc/ServicePageGenerator.cs [hunk 4]
```diff
@@ -122,18 +164,6 @@ public void GenerateServicePageFromTopLevelTOC(TableOfContentsNode node, List<Fi
...
-                    GenerateServicePageFromTopLevelTOC(item, results, $"{directoryName}");
-                }
-                else
-                {
-                    GenerateServicePageFromTopLevelTOC(item, results, $"{directoryName}/{filename}");
-                }
-            }
         }
 
         private string? GetHrefRelativeToServicePage(string childHref, string referenceTOCFullPath, FilePath servicePagePath)
```
Comment (live1206) [inline src/docfx/build/toc/ServicePageGenerator.cs:169]: 'referenceTOCFullPath' not used in the method?

File: src/docfx/build/toc/ServicePageGenerator.cs (hunk 1 above)
Comment (AngryBerryMS) [reply src/docfx/build/toc/ServicePageGenerator.cs]: Fixed。

**Review by live1206 — APPROVED** [review]

**Pull request merged**